polkadot/zombienet-sdk-tests/tests/helpers/mod.rs

```diff
@@ -79,3 +80,25 @@ pub async fn assert_finalized_block_height(
 	}
 	Ok(())
 }
+
+/// Assert that finality has not stalled.
+pub async fn assert_blocks_are_being_finalized(
+	client: &OnlineClient<PolkadotConfig>,
+) -> Result<(), anyhow::Error> {
+	let mut finalized_blocks = client.blocks().subscribe_finalized().await?;
+	let first_measurement = finalized_blocks
+		.next()
+		.await
+		.ok_or(anyhow::anyhow!("Can't get finalized block from stream"))??
+		.number();
+	sleep(Duration::from_secs(12)).await;
+	let second_measurement = finalized_blocks
```

Comment:
Sorry for being hypocritical as I have written such checks in my test as well but if the finality has stalled won't we block here forever (or at least when a new block is finalized)?

Reply:
Yeah, I wasn't sure about sprinkling timeouts everywhere, but added a timeout on the call site as it makes sense.